src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Gateway;
+using Kmd.Logic.Identity.Authorization;
+using Microsoft.Rest;
+using YamlDotNet.Serialization;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    [SuppressMessage("Design", "CA1001:Types that own disposable fields should be disposable", Justification = "HttpClient is not owned by this class.")]
+    public class Publish : IPublish
+    {
+        private readonly HttpClient httpClient;
+        private readonly GatewayOptions options;
+        private readonly LogicTokenProviderFactory tokenProviderFactory;
+        private IGatewayClient gatewayClient;
+        private IList<PublishResult> publishResults;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Publish"/> class.
+        /// </summary>
+        /// <param name="httpClient">The HTTP client to use. The caller is expected to manage this resource and it will not be disposed.</param>
+        /// <param name="tokenProviderFactory">The Logic access token provider factory.</param>
+        /// <param name="options">The required configuration options.</param>
+        public Publish(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, GatewayOptions options)
+        {
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+            this.tokenProviderFactory = tokenProviderFactory ?? throw new ArgumentNullException(nameof(tokenProviderFactory));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (string.IsNullOrEmpty(this.tokenProviderFactory.DefaultAuthorizationScope))
+            {
+                this.tokenProviderFactory.DefaultAuthorizationScope = "https://logicidentityprod.onmicrosoft.com/bb159109-0ccd-4b08-8d0d-80370cedda84/.default";
+            }
+
+            this.publishResults = new List<PublishResult>();
+#pragma warning restore CS0618 // Type or member is obsolete
+        }
+
+        /// <summary>
+        /// Create gateway entities.
+        /// </summary>
+        /// <param name="folderPath">Folder path provider all gateway entries details.</param>
+        /// <returns>Error details on failure, gateway entities name on success.</returns>
+        public async Task<IList<PublishResult>> ProcessAsync(string folderPath)
```

Comment:
Result should be rather read only. Set it to IReadOnlyList or IEnumerable. Whatever suits you more.